api/ocm/resolver.go

```diff
@@ -86,3 +147,5 @@ type MatchingResolver interface {
 func NewMatchingResolver(ctx ContextProvider) MatchingResolver {
 	return internal.NewMatchingResolver(ctx.OCMContext())
 }
+
+type ResolverRule = internal.ResolverRule
```

Comment:
Wouldn't it make more sense to put it in here: `api/ocm/interface.go` or `api/ocm/types/interface.go`?

By the way: What's the purpose of 'api/ocm/types/interface.go'? The types are the same as in 'api/ocm/interface.go'! Why do we need this code duplication?

Reply:
The resolver.go contains some standard resolver implementations. The matching resolver implementation must be in internal, because it is used by the context, therefore it has to be forwarded. To keep all the stuff together, it is in the resolver source file.